stdlib/test/random/test_random.mojo

```diff
@@ -73,6 +80,94 @@ def test_seed():
     assert_equal(some_unsigned_integer, random_ui64(0, 255))
 
 
+def test_shuffle():
+    # TODO: Clean up with list comprehension when possible.
+
+    # Property tests
+    alias L_i = List[Int]
+    alias L_s = List[String]
+    var a = L_i(1, 2, 3, 4)
+    var b = L_i(1, 2, 3, 4)
+    var c = L_s("Random", "shuffle", "in", "Mojo")
+    var d = L_s("Random", "shuffle", "in", "Mojo")
+
+    shuffle(b)
+    assert_equal(len(a), len(b))
+    assert_true(a != b)
+    for i in range(len(b)):
+        assert_true(a.__contains__(b[i]))
```

Comment:
**Question** Can we safely use the `in` sugar syntax since this test isn't interested in testing dunder contains directly?